src/modules/home/components/Postings.js

```diff
@@ -2,38 +2,54 @@ import React, { useEffect, useState } from 'react'
 import { Post } from './Post'
 import axios from 'axios'
 import styles from '../styles/Postings.module.css'
+import { LoadingSpinner } from '../../common/components/LoadingSpinner'
 
 export const Postings = () => {
     const [posts, setPosts] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
 
     useEffect(() => {
-        axios
-            // eslint-disable-next-line no-undef
-            .get(`${process.env.REACT_APP_SERVICE_ENDPOINT}/post/getPosts`)
-            .then((res) => {
-                setPosts(res.data.posts)
-            })
-            .catch((err) => console.log(err))
-    }, [])
-
-    if (!posts || !posts.length) {
-        return (
-            <div className={styles.posting}>No posts yet, check back later</div>
-        )
-    }
+        const getPosts = async () => {
+            setIsLoading(true)
+            try {
+                await axios
+                    .get(
+                        // eslint-disable-next-line no-undef
+                        `${process.env.REACT_APP_SERVICE_ENDPOINT}/post/getPosts`
+                    )
+                    .then((res) => {
```

Comment:
remoe the .then it doesnt do anything now that we are using await
await axios
                    .get(
                        // eslint-disable-next-line no-undef
                        `${process.env.REACT_APP_SERVICE_ENDPOINT}/post/getPosts`
                    )
setPosts(res.data.posts)
setIsLoading(false)
  } catch (error) {
                console.log(error)
            }

Reply:
done, but had undefined error, so had to define response